Assets/Scripts/Enrico/Inventory.cs

```diff
@@ -1,66 +1,66 @@
-                string json = i.CustomData;                                                 // Store cloudData into a string
-                File.WriteAllText(Application.dataPath + "/savedJson.json", json);         //Write a json file to assets
-                string read = File.ReadAllText(Application.dataPath + "/savedJson.json"); //Read from json file and store it
-                JsonDataHolder jsonSet = JsonUtility.FromJson<JsonDataHolder>(read);     // Apply the read data and set the data the json class.
-                ////////////////////////////////////////////////
-                
-                foreach (var info in allResources.resourceInfo)
-                {
-                    if (info.resourceName == i.DisplayName)
-                    {
-                        info.buildingCost = (int)cost;
-                        
-                        ////Apply Json Values to the player.
-                        if(jsonSet != null)
-                        {
-                            info.damage = jsonSet.Damage;
-                        }
-                        Debug.Log("cost " + cost + " name " + i.DisplayName);
-                    }
-                    
-                }
-            }
-        }, error =>
-        {
-            
-        });
-    }
-}
-
-    
+        }
+        GetCatalogItemsRequest catalogItems = new GetCatalogItemsRequest();
+        catalogItems.CatalogVersion = "Upgrades";
+
+        PlayFabClientAPI.GetCatalogItems(catalogItems, result =>
+        {
+            List<CatalogItem> items = result.Catalog;
+           // result.CustomData //////////////////////////////////TO TEST WITH THIS and with more keys per item
+            foreach (CatalogItem i in items)
+            {
+                uint cost = i.VirtualCurrencyPrices["SL"];
+                
+                ///////JSON STUFF HERE/////////////////////////////
+                string json = i.CustomData;                                                 // Store cloudData into a string
+                File.WriteAllText(Application.dataPath + "/savedJson.json", json);         //Write a json file to assets
+                string read = File.ReadAllText(Application.dataPath + "/savedJson.json"); //Read from json file and store it
+                JsonDataHolder jsonSet = JsonUtility.FromJson<JsonDataHolder>(read);     // Apply the read data and set the data the json class.
+                ////////////////////////////////////////////////
+                
+                foreach (var info in allResources.resourceInfo)
+                {
+                    if (info.resourceName == i.DisplayName)
+                    {
+                        info.buildingCost = (int)cost;
+                        
+                        ////Apply Json Values to the player.
+                        if(jsonSet != null)
+                        {
+                            info.damage = jsonSet.Damage;
+                        }

```

Comment:
Can single line this

Reply:
True, left cos we need to add more. true tho